synapse/cores/lmdb.py

```diff
@@ -515,7 +640,9 @@ def _getSizeByProp(self, prop, valu=None, limit=None, mintime=None, maxtime=None
 
     def _getRowsByProp(self, prop, valu=None, limit=None, mintime=None, maxtime=None,
                        do_count_only=False):
-        indx = self.index_pt if valu is None else self.index_pvt
+        indx = self._dbs.get(PROP_VAL_INDEX)
```

Comment:
In my own testing, this design pattern more than double the time it takes to do a basic retrieval. https://gist.github.com/vEpiphyte/73c58ce761bf82bde58ab300ff3b386b